src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceHelpers.cs

```diff
@@ -96,28 +103,244 @@ public static async Task<Location> GetLocationInGeneratedSourceAsync(SymbolKey s
             return null;
         }
 
-        public static bool TryGetImplementationAssemblyPath(string referenceDllPath, [NotNullWhen(true)] out string? implementationDllPath)
+        /// <summary>
+        /// Uses various heuristics to try to find the implementation assembly for a reference assembly without
+        /// loading 
+        /// </summary>
+        public static bool TryFindImplementationAssemblyPath(string referencedDllPath, [NotNullWhen(true)] out string? implementationDllPath)
```

Comment:
>TryFindImplementationAssemblyPath [](http://example.com/codeflow?start=27&length=33)

Shouldn't this be async and read files using async API?

Reply:
Sure, why not ¯\\\_(ツ)_/¯

Though I wouldn't expect much difference. This only reads one XML file, and only the first line of that file. In fact, reading a file that is just `<FileList FrameworkName="MyPack">` doesn't even throw an XmlException, because it doesn't read far enough to determine that it's invalid :)

Reply:
Actually.. because of the out parameter, this is a bit annoying. Is it really going to make any meaningful difference? Only the `Read()` call becomes `ReadAsync()`, there is nothing else that is async.